src/Common/tests/Tests/System/StringTests.cs

```diff
@@ -7398,11 +7398,12 @@ public static void InternalTestAotSubset()
             Assert.True(s.IsNormalized(NormalizationForm.FormD));
             Assert.True(s.IsNormalized(NormalizationForm.FormKC));
             Assert.True(s.IsNormalized(NormalizationForm.FormKD));
-            Assert.Same(s, s.Normalize());
-            Assert.Same(s, s.Normalize(NormalizationForm.FormC));
-            Assert.Same(s, s.Normalize(NormalizationForm.FormD));
-            Assert.Same(s, s.Normalize(NormalizationForm.FormKC));
-            Assert.Same(s, s.Normalize(NormalizationForm.FormKD));
+
+            Assert.Equal(s, s.Normalize());
+            Assert.Equal(s, s.Normalize(NormalizationForm.FormC));
+            Assert.Equal(s, s.Normalize(NormalizationForm.FormD));
+            Assert.Equal(s, s.Normalize(NormalizationForm.FormKC));
+            Assert.Equal(s, s.Normalize(NormalizationForm.FormKD));
```

Comment:
This test tests only CoreCLR logic (to return the same instance): https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/String.cs#L798-L808

Reply:
How about something like, pseudo-code:
```C#
void AssertEqualOrSame(string expected, string actual) =>
    PlatformDetection.IsMonoRuntime ? Assert.Equal(expected, actual) : Assert.Same(expected, actual);
AssertEqualOrSame(s, s.Normalize(NormalizationForm.Formc));
...
```
?

Reply:
This one specifically should be guarded by something like `if (PlatformDetection.IsCoreClr)` (which doesn't exist AFAIK).

Reply:
Yeah I wanted to add `public static bool IsCoreClrRuntime => IsNetCore && !IsMonoRuntime;` for that